### PR TITLE
juju client displays space name

### PR DESCRIPTION
**Pull request by hmlanigan**
#10251  Description of change

Ensure juju client output displays space names, not space IDs.  Updates to both show-application and status for this to happen.  

model.AllEndpoints() structure returns a Binding object to facilitate the above status update.

## QA steps

Deploy application with bindings, bound to a space:  
1. `juju show-application` should display the endpoint bindings with the space names.
2. `juju status --format yaml` should display the endpoint bindings with the space names.

### CODE REVIEW

File: apiserver/facades/client/application/application.go [hunk 1]
```diff
@@ -2397,9 +2397,12 @@ func (api *APIBase) ApplicationsInfo(in params.Entities) (params.ApplicationInfo
 			continue
 		}
 
-		// TODO (hml) 2019-10-16
-		// bindings.MapWithSpaceNames() might be of
-		// use here.
```
Comment (SimonRichardson) [inline apiserver/facades/client/application/application.go]: Indeed :D

File: state/model_test.go [hunk 2]
```diff
@@ -560,20 +560,19 @@ func (s *ModelSuite) TestAllEndpointBindings(c *gc.C) {
...
+		"foo-bar":         network.DefaultSpaceId,
+		"db-client":       network.DefaultSpaceId,
+		"admin-api":       network.DefaultSpaceId,
+		"url":             network.DefaultSpaceId,
+		"logging-dir":     network.DefaultSpaceId,
+		"monitoring-port": network.DefaultSpaceId,
+		"db":              oneSpace.Id(),
+	}
+	c.Assert(listBindings[0].AppName, gc.Equals, app.Name())
+	c.Assert(listBindings[0].Bindings.Map(), gc.DeepEquals, expected)
```
Comment (SimonRichardson) [inline state/model_test.go]: Should we have a `MapWithSpaceNames` test or do we have one already here?

Comment (hmlanigan) [reply state/model_test.go]: have one for the Bindings object: bindingsMockSuite.TestMapWithSpaceNames()

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
$$merge$$

intermittent fail in IDWatcherSuite.TestWatcher

**Pull request merged**